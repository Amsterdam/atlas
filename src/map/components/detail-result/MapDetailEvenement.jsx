import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultItem from './MapDetailResultItem';
import MapDetailResultWrapper from './MapDetailResultWrapper';

const MapDetailEvenement = ({ panoUrl, item, onMaximize, onPanoPreviewClick }) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={item.label}
    title="Evenement"
  >
    <ul className="map-detail-result__list">
      <MapDetailResultItem
        label="Titel"
        value={item.titel}
      />
      <MapDetailResultItem
        label="Startdatum"
        value={item.startdatum}
      />
      <MapDetailResultItem
        label="Einddatum"
        value={item.einddatum}
      />
      <MapDetailResultItem
        label="Website"
        value={item.url}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailEvenement.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string,
    url: PropTypes.string
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailEvenement;
