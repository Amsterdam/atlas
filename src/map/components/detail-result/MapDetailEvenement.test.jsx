import React from 'react'
import { shallow } from 'enzyme'

import MapDetailEvenement from './MapDetailEvenement'

describe('MapDetailEvenement', () => {
  it('should render everything', () => {
    const evenement = {
      titel: 'Bedrijfsinvesteringszone heffingLabel',
      startdatum: '12-12-2012',
      einddatum: '12-12-2012',
      url: 'https://evenement/link',
      label: 'Evenement label',
    }
    const clickHandler = jest.fn()
    const wrapper = shallow(
      <MapDetailEvenement
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        item={evenement}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
