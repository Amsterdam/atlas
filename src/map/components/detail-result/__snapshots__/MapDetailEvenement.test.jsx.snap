// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailEvenement should render everything 1`] = `
<Connect(MapDetailResultWrapper)
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Evenement label"
  title="Evenement"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Startdatum"
      value="12-12-2012"
    />
    <MapDetailResultItem
      label="Einddatum"
      value="12-12-2012"
    />
    <MapDetailResultUrlItem
      description="https://evenement/link"
      label="Meer informatie"
      link="https://evenement/link"
    />
  </ul>
</Connect(MapDetailResultWrapper)>
`;
